File: ABC/ABC301-400/ABC341/C/C.py
H, W, N = map(int, input().split())
T = input()
S = [input() for _ in range(H)]
ans = 0
for i in range(H):
	for j in range(W):
		if S[i][j] == '#':
			continue
		nowx, nowy = i, j
		ok = True
		for c in T:
			if c == 'U':
				nowx -= 1
			if c == 'D':
				nowx += 1
			if c == 'L':
				nowy -= 1
			if c == 'R':
				nowy += 1
			if S[nowx][nowy] == '#':
				ok = False
				break
		if ok:
			ans += 1
print(ans)

File: ABC/ABC301-400/ABC341/C/C.nim
import strutils, sequtils
let h_w_n = stdin.readLine.split().map(parseInt)
let
  H = h_w_n[0]
  W = h_w_n[1]
  N = h_w_n[2]
  T = stdin.readLine
var S = newSeq[string](H)
for i in 0..<H:
  S[i] = stdin.readLine
var ans = 0
for i in 0..<H:
  for j in 0..<W:
    if S[i][j] == '#':
      continue
    var
      nowx = i
      nowy = j
      ok = true
    for c in T:
      if c == 'U':
        dec nowx
      if c == 'D':
        inc nowx
      if c == 'L':
        dec nowy
      if c == 'R':
        inc nowy
      if S[nowx][nowy] == '#':
        ok = false
        break
    if ok:
      inc ans
echo ans